dandi/tests/test_metadata.py

```diff
@@ -127,7 +126,7 @@ def test_metadata2asset_simple1(schema_dir):
             "nwb_version": "2.2.5",
             "experiment_description": "experiment_description1",
             "experimenter": ("experimenter1",),
-            "identifier": "bfc23fb6192b41c083a7257e09a3702b",
+            "id": "urn:uuid:bfc23fb6192b41c083a7257e09a3702b",
```

Comment:
seems to be not a uuid here.

But I also thought that a brief discussion on monday just boiled down to having `identifier` with actual value and `id` been either
- minted/populated by dandi-api when returning an adjusted metadata record.  Thus having both, and thus resulting in duplication.  
- or just produced while preparing json-ld view of the metadata

without that -- for any use of this `id` as an `asset_id` in the API endpoints a programmer would need to first strip this `urn:uuid:` prefix (we do not want `:` in the path - would need to be url encoded), we would loose clear mapping from an `asset_id` to its location in the key store . Altogether sounds like more pain IMHO for no gain.
Thus I still have original question on what problem (exactly) you are trying to solve with such `identifier` -> `id` replacement?  
if it is just to produce json-ld ready metadata, wouldn't such records be tuned anyways and accompanied with context etc?  or
do you see us using some other than `urn:uuid:` prefix for the assets?

Reply:
that is correct. i haven't added any validation here. 

regarding id vs identifier this is mimicing the schema.org setup. `id` maps to a URI (i.e. an identity for the object), while `identifier` is not necessarily a URI (it could be simply a text. when producing a context, i need to be able to map properties of a class to types. previously i was mapping identifier to id and therefore a URI, but that meant wherever i see identifier (which was in every class, it would automatically be turned into a uri, which did not make sense for certain things like orcid).

in this formulation, we explicitly ensure that value of id for an asset is a urn, which means know expansion needs to take place. it's an identifier that can be used in jsonld, rdf, etc.,. whereas if it was a uuid, it would require being turned into a uri. 

regarding parsing by library, i don't think that's a pain: this is a fully formed uri so urlparse and other parsers will work out of the box. also i would really want the cli to simply use the models when dealing with metadata from the api, so technically that should work as well.